### PR TITLE
dogstatsd: remove stale UNIX socket on startup

### PR DESCRIPTION
**Pull request by j-vizcaino**
dogstatsd removes stale UNIX socket when starting. This logic only applies to existing UNIX socket. If anything else (file, directory, etc...) exists for the same path, the `NewUDSListener()` function fails and returns an error.

Entrypoint scripts for the agent and dogstatsd standalone image have been updated.

### CODE REVIEW

File: pkg/dogstatsd/listeners/uds_common_test.go
Comment (mfpierre) [inline pkg/dogstatsd/listeners/uds_common_test.go]: This order was actually good, we group by native/vendor/project dependencies

File: releasenotes/notes/fix-dsd-stale-unix-socket-542e6b1367e45d35.yaml
Comment (mfpierre) [inline releasenotes/notes/fix-dsd-stale-unix-socket-542e6b1367e45d35.yaml]: I think it could be listed under `enhancements` instead of `fixes`:

> the clean-up of the UNIX socket is now done directly by dogstatsd instead of the image entrypoint

**Review by mfpierre — CHANGES_REQUESTED** [review]
@j-vizcaino thanks for the PR 🙇 

LGTM overall, left a few nits

File: pkg/dogstatsd/listeners/uds_common_test.go
Comment (j-vizcaino) [reply pkg/dogstatsd/listeners/uds_common_test.go]: Damn, GoLand took care of this for me. Maybe I'm missing some option to have the import grouped according this order.
Sorry about the noise.

**Comment by j-vizcaino** [comment]
@mfpierre Fixed! Should be good to go. Thanks!

**Review by mfpierre — DISMISSED** [review]
LGTM

**Review by olivielpeau — DISMISSED** [review]
👌 just a couple of typos to fix, feel free to merge once fixed

**Review by olivielpeau — APPROVED** [review]
:shipit:

**Comment by j-vizcaino** [comment]
@mfpierre waiting on your stamp of approval ;)

**Review by mfpierre — APPROVED** [review]

**Pull request merged**
